packages/adapter-surrealdb/src/index.ts

```diff
@@ -0,0 +1,364 @@
+import type Surreal from "surrealdb.js"
+import type {
+  Adapter,
+  AdapterUser,
+  AdapterAccount,
+  AdapterSession,
+  VerificationToken,
+} from "next-auth/adapters"
+import type { ProviderType } from "next-auth/providers"
+
+type Document = Record<string, string | null | undefined> & { id: string }
+export type UserDoc = Document & { email: string }
+export type AccountDoc<T = string> = {
+  id: string
+  userId: T
+  refresh_token?: string
+  access_token?: string
+  type: ProviderType
+  provider: string
+  providerAccountId: string
+  expires_at?: number
+}
+export type SessionDoc<T = string> = Document & { userId: T }
+
+const extractId = (surrealId: string) => surrealId.split(":")[1] ?? surrealId
```

Comment:
@martinschaer Any reason why you prefer to extract the id? This makes the query on the server side having to append the pre-split values.

Reply:
@leolorenzoluis just to hide implementation details for security, so that if a user sees an ID its not that obvious what kind of DB its running below